lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -531,6 +538,26 @@ tokenize(String, Line, Column, Scope, Tokens) ->
       {error, Reason, String, Tokens}
   end.
 
+maybe_warn_for_ambiguous_bang_before_equals(Kind, Atom, [$= | _], Scope, Line) ->
+  {What, Identifier} =
+    case Kind of
+      atom -> {"atom", [$: | atom_to_list(Atom)]};
+      identifier -> {"identifier", atom_to_list(Atom)}
+    end,
+
+  case lists:last(Identifier) of
+    Last when Last == $!; Last == $? ->
+      Msg = io_lib:format("found an ~ts \"~ts\", ending with ~ts, followed by =. "
```

Comment:
We need to remove the article "an", since we are specifying the atom/identifier, and I would wrap `ending with ~ts` as `ending with "~ts"`.

Reply:
Good catches, pushed!